cibyl/sources/elastic/client.py

```diff
@@ -0,0 +1,72 @@
+"""
+#    Copyright 2022 Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+
+import logging
+from elasticsearch import Elasticsearch
+from elasticsearch.exceptions import ConnectionError, ConnectionTimeout
+from sys import exit
+
+
+LOG = logging.getLogger(__name__)
+
+
+class ElasticSearchClient:
+
+    def __init__(self: object,
+                 host: str = "http://localhost",
+                 port: int = 9200) -> None:
+        self.host = f"{host}:{port}"
+
+    def connect(self: object) -> Elasticsearch:
+        """This is used for connecting to the ElasticSearch instance.
+
+        :param host: Elasticsearch host including the scheme
+        :type host: str, optional
+        :param port: Port for sending REST requests to the instance
+        :type port: str, optional
+        :return: Elasticsearch module instance
+        :rtype: Elasticsearch
+        :raises: ConnectionTimeout:
+                 If the connection attempt reach the max time
+        :raises: ConnectionError:
+                 If exists an handled connection error
+        :raises: Exception:
+                 If exists an unhandled connection error
+        """
+        try:
+            es = Elasticsearch(self.host)
+            es.ping()
+        except ConnectionTimeout as e:
+            LOG.error(f"Timeout connection to ElasticSearch. Details: {e}")
+            exit(1)
+        except ConnectionError as e:
+            LOG.error(f"Connection error to ElasticSearch. Details: {e}")
+            exit(1)
+        except Exception as e:
+            LOG.error(f"An unknown error occurred connecting to ElasticSearch. \
+                        Details: {e}")
+            exit(1)
```

Comment:
My advice for error handling is that you let the caller catch the error and react to it. Quitting the app without any chance for recovery reduces our flexibility, imagine for example that later on we want to retry the connection a couple of times before giving up. That could be done by the caller and we would not need to make changes here. In addition, it would match the documentation of the method as right now it is not raising any of them.

Reply:
To be exquisite, you could create a single ElasticSearchError class and raise only that one. I do not think that differentiating between the errors does much for us, we just cannot connect to the database.

Reply:
Good advice. I'll change it. Thx :+1:

Reply:
`ElasticSearchError` class added as exception.